src/autolabel/labeling_output.py

```diff
@@ -0,0 +1,211 @@
+        self.task = TaskFactory.from_config(self.config)
+
+        if llm_labels is not None:
+            self.add_columns_to_df(llm_labels)
+
+    def add_columns_to_df(self, llm_labels: List[LLMAnnotation]):
+        # Add the LLM labels to the dataframe
+        self.df[self.prefix + "label"] = [x.label for x in llm_labels]
+
+        # Add the LLM errors to the dataframe
+        self.df[self.prefix + "error"] = [x.error for x in llm_labels]
+
+        # Add labeled success column to the dataframe
+        self.df[self.prefix + "successfully_labeled"] = [
+            x.successfully_labeled for x in llm_labels
+        ]
+
+        self.df[self.prefix + "annotation"] = [x.json() for x in llm_labels]
+
+        # Add row level LLM metrics to the dataframe
+        if self.metrics is not None:
+            for metric in self.metrics:
+                if (
+                    isinstance(metric.value, list)
+                    and len(metric.value) == self.df.shape[0]
+                ):
+                    self.df[self.prefix + metric.name] = metric.value
+
+        # Add the LLM confidence scores to the dataframe if confidence is set in config
+        if self.config.confidence():
+            self.df[self.prefix + "confidence"] = [
+                x.confidence_score for x in llm_labels
+            ]
+
+        # Add the LLM explanations to the dataframe if chain of thought is set in config
+        if self.config.chain_of_thought():
+            self.df[self.prefix + "explanation"] = [l.explanation for l in llm_labels]
+
+    def __repr__(self):
+        if self.df is not None:
+            return self.df.__repr__()
+
+    def __str__(self):
+        if self.df is not None:
+            return self.df.__str__()
+
+    def save(self, output_file_name: str):
+        if output_file_name.endswith(".csv"):
+            self.df.to_csv(
+                str(output_file_name),
+                sep=self.config.delimiter(),
+                header=True,
+                index=False,
+            )
+        elif output_file_name.endswith(".jsonl"):
+            self.df.to_json(
+                str(output_file_name),
+                orient="records",
+                lines=True,
+                force_ascii=False,
+            )
+        else:
+            raise ValueError(f"Unsupported output file format: {output_file_name}")
+
+    def filter(self, label=None, ground_truth=None, filter_func=None):
+        filtered_df = self.df
+
+        if label:
+            filtered_df = filtered_df[filtered_df[self.prefix + "label"] == label]
+
+        if ground_truth:
+            filtered_df = filtered_df[
+                filtered_df[self.config.label_column()] == ground_truth
+            ]
+
+        if filter_func:
+            filtered_df = filtered_df.apply(filter_func, axis=1)
+
+        return LabelingOutput(
+            self.config,
+            filtered_df,
+            self.metrics,
+        )
+
+    def errors(self):
```

Comment:
errors vs mistakes is a bit confusing naming - how about we roll it into the same function? 

llm label != gt label is ultimately a kind of error?

Reply:
I wanted to separate them out in order to find the examples where the model was not able to answer, if potentially they went over context length or some other input specific problem and we wanted to fix them